### PR TITLE
Preselect sms option

### PR DESCRIPTION
**Pull request by seromenho**
We can now send type to this provider so it opens the ds query provider
with specific options.

And prepare for domain white listing

### CODE REVIEW

File: js/interface.js [hunk 1]
```diff
@@ -48,38 +48,54 @@ function selectDataSource(ds) {
...
-    modes: [{
-        label: 'By email',
-        filters: false,
-        columns: [{
-          key: 'emailMatch',
+var dsQueryData = {};
+if (data.type === 'sms') {
+  dsQueryData = {
+    settings: {
+      dataSourceLabel: 'Select a data source',
```
Comment (tonytlwu) [inline js/interface.js]: 'Select the data source containing the user information'

File: js/interface.js [hunk 2]
```diff
@@ -114,21 +130,49 @@ var dsQueryProvider = Fliplet.Widget.open('com.fliplet.data-source-query', {
...
+    case 'email':
+      validation.email = {
+        toColumn: result.data.columns.emailMatch,
+        matchColumn: result.data.columns.emailMatch,
+        template: emailProviderResult || defaultEmailSettings,
+        expire: $('#expire-timeout').val(),
+      }
+      break;
+
+    case 'domain':
```
Comment (tonytlwu) [inline js/interface.js:160]: You wouldn't get a `domain` type validation. Domain whitelisting is a `email` type validation with domain whitelisting enabled as an additional feature.

e.g. Search everything in the datasource for the user's email, and also allow it if it it's in one of these domains.

**We should resolve this later though. I don't want to delay the launch of SMS verification by even a minute.**

File: js/interface.js [hunk 3]
```diff
@@ -138,9 +182,11 @@ dsQueryProvider.then(function onForwardDsQueryProvider(result) {
...
   Fliplet.DataSources.update(options)
     .then(function() {
-      data = {
-        type: result.data.selectedModeIdx === 1 ? 'sms' : 'email',
-        dataSourceQuery: result.data
+      data.dataSourceQuery = result.data;
+
+      // Email verification type is choosed on ds query provider
+      if (data.type !== 'sms') {
+        data.type = result.data.selectedModeIdx === 1 ? 'domain' : 'email';
```
Comment (tonytlwu) [inline js/interface.js:189]: I don't think we need this any more.

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: js/interface.js (hunk 2 above)
Comment (seromenho) [reply js/interface.js:160]: Somehow I need to store the settings under this setting on the data source definition...
We need to make it like that so when you are requesting the code to the server we know if you are using just email or domain. They don't work in the same way on the server.

File: js/interface.js (hunk 3 above)
Comment (seromenho) [reply js/interface.js:189]: We do in my point of view.

Comment (seromenho) [reply js/interface.js:189]: When the build calls the send-validation endpoint it needs to know if it's regular validation or with email domain white listing.

File: js/interface.js (hunk 1 above)
Comment (seromenho) [reply js/interface.js]: Done.

**Review by tonytlwu — APPROVED** [review]
@seromenho We can discuss how this works for email verification later. The SMS part looks good to me. Make sure we have a chat before you start working on email verification + domain whitelisting.

**Pull request merged**